internal/provider/dedicated_network_container_data_source.go

```diff
@@ -0,0 +1,119 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+type dedicatedNetworkContainerDataSourceData struct {
```

Comment:
since network container is a project level resource, I suggest adding a project_id attribute it. Not just use labels.